doc/modules/neighbors.rst

```diff
@@ -514,3 +514,224 @@ the model from 0.81 to 0.82.
+classification in the embedded space. Here is an example using the two classes:
+
+    >>> from sklearn.neighbors import NeighborhoodComponentsAnalysis
+    >>> from sklearn.neighbors import KNeighborsClassifier
+    >>> from sklearn.datasets import load_iris
+    >>> from sklearn.model_selection import train_test_split
+    >>> X, y = load_iris(return_X_y=True)
+    >>> X_train, X_test, y_train, y_test = train_test_split(X, y,
+    ... stratify=y, test_size=0.7, random_state=42)
+    >>> nca = NeighborhoodComponentsAnalysis(random_state=42)
+    >>> nca.fit(X_train, y_train) # doctest: +ELLIPSIS
+    NeighborhoodComponentsAnalysis(...)
+    >>> # Apply the learned transformation when using KNeighborsClassifier
+    >>> knn = KNeighborsClassifier(n_neighbors=3)
+    >>> knn.fit(nca.transform(X_train), y_train) # doctest: +ELLIPSIS
+    KNeighborsClassifier(...)
+    >>> print(knn.score(nca.transform(X_test), y_test))
+    0.961904761905
+
+Alternatively, one can create a :class:`sklearn.pipeline.Pipeline` instance
+that automatically applies the transformation when fitting or predicting:
+
+    >>> from sklearn.pipeline import Pipeline
+    >>> nca = NeighborhoodComponentsAnalysis(random_state=42)
+    >>> knn = KNeighborsClassifier(n_neighbors=3)
+    >>> nca_pipe = Pipeline([('nca', nca), ('knn', knn)])
+    >>> nca_pipe.fit(X_train, y_train) # doctest: +ELLIPSIS
+    Pipeline(...)
+    >>> print(nca_pipe.score(X_test, y_test))
+    0.961904761905
+
+.. |nca_classification_1| image:: ../auto_examples/neighbors/images/sphx_glr_plot_nca_classification_001.png
+   :target: ../auto_examples/neighbors/plot_nca_classification.html
+   :scale: 50
+
+.. |nca_classification_2| image:: ../auto_examples/neighbors/images/sphx_glr_plot_nca_classification_002.png
+   :target: ../auto_examples/neighbors/plot_nca_classification.html
+   :scale: 50
+
+.. centered:: |nca_classification_1| |nca_classification_2|
+
+The plot shows decision boundaries for Nearest Neighbor Classification and
+Neighborhood Components Analysis classification, when training and scoring
+on only two features, for visualisation purpose.
+
+
+Dimensionality reduction
+------------------------
+
+:class:`NeighborhoodComponentsAnalysis` can be used to perform supervised
+dimensionality reduction. The input data are projected onto a linear subspace
+consisting of the directions which minimize the NCA objective. The desired
+dimensionality can be set using the parameter ``n_features_out``. For instance,
+the following shows a comparison of dimensionality reduction with Principal
+Component Analysis (:class:`sklearn.decomposition.PCA`), Linear Discriminant
+Analysis (:class:`sklearn.discriminant_analysis.LinearDiscriminantAnalysis`)
+and Neighborhood Component Analysis (:class:`NeighborhoodComponentsAnalysis`)
+on the Digits dataset, a dataset with size :math:`n_{samples} = 1797` and
+:math:`n_{features} = 64`. The data set is splitted in a training and test set
```

Comment:
is split into

also I would maybe remove the next sentence and simplify to something like ""The dataset is spit into a training and a test set of equal size, then standardized"

Reply:
Done.